addon/ng2/blueprints/route/index.js

```diff
@@ -38,9 +38,16 @@ module.exports = {
 
   normalizeEntityName: function (entityName) {
     var parsedPath = dynamicPathParser(this.project, entityName);
-
+    
+    // if user specified a route starts w/ `+` remove it as it'd break convention
+    if (parsedPath.name.indexOf('+') === 0) {
+      // parsedPath.name = parsedPath.name.substr(1);
+      var index = entityName.lastIndexOf(parsedPath.name);
+      entityName = entityName.substr(0, index) + entityName.substr(index + 1);
```

Comment:
Could that work?

`entityName = path.dirname(entityName) + path.basename(entityName).substr(1);`